server/src/main/java/org/elasticsearch/common/unit/ByteSizeValue.java

```diff
@@ -123,15 +124,24 @@ public static ByteSizeValue readFrom(StreamInput in) throws IOException {
 
     @Override
     public void writeTo(StreamOutput out) throws IOException {
-        TransportVersion tv = out.getTransportVersion();
-        if (tv.onOrAfter(BYTE_SIZE_VALUE_ALWAYS_USES_BYTES_1) && tv.before(REVERT_BYTE_SIZE_VALUE_ALWAYS_USES_BYTES_1)) {
+        if (alwaysUseBytes(out.getTransportVersion())) {
             out.writeZLong(sizeInBytes);
         } else {
             out.writeZLong(Math.divideExact(sizeInBytes, desiredUnit.toBytes(1)));
         }
         desiredUnit.writeTo(out);
     }
 
+    private static boolean alwaysUseBytes(TransportVersion tv) {
+        if (tv.between(BYTE_SIZE_VALUE_ALWAYS_USES_BYTES_1, REVERT_BYTE_SIZE_VALUE_ALWAYS_USES_BYTES_1)) {
```

Comment:
Typically we would chain these in a single expression, no need to do jumps. And prefer the "expected", so start with the onOrAfter, then isPatchFrom, then the between.